core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -0,0 +1,223 @@
+package webapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk"
+	wasmpb "github.com/smartcontractkit/chainlink-common/pkg/workflows/wasm/pb"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/validation"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/capabilities"
+	ghcapabilities "github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/common"
+)
+
+var _ connector.GatewayConnectorHandler = &OutgoingConnectorHandler{}
+
+type OutgoingConnectorHandler struct {
+	gc            connector.GatewayConnector
+	method        string
+	lggr          logger.Logger
+	responseChs   map[string]chan *api.Message
+	responseChsMu sync.Mutex
+	rateLimiter   *common.RateLimiter
+	generateUUID  func() uuid.UUID
+}
+
+// Config is the configuration for OutgoingConnectorHandler.
+// Currently used by the WebApi Target and Compute Action capability & handler
+// TODO: handle retry configurations here CM-472
+// Note that workflow executions have their own internal timeouts and retries set by the user
+// that are separate from this configuration
+type Config struct {
+	UUIDGeneratorFn func() uuid.UUID
```

Comment:
@agparadiso The fact that this is a UUID is arguably an implementation detail, consider just calling this IDGenerator and returning a string directly -- you're going to stringify it at some point anyway, so may as well do that here to hide the type.

Reply:
(This also makes mocking easier)